girder/cumulus/server/volume.py

```diff
@@ -267,7 +255,7 @@ def attach(self, volume, cluster, params):
         p = CloudProvider(dict(secretAccessKey=secret_key, **profile))
 
         aws_volume = p.get_volume(volume)
-        if aws_volume['state'] != VolumeState.AVAILABLE:
+        if aws_volume and aws_volume['state'] != VolumeState.AVAILABLE:
```

Comment:
At first glance the logic here looks wrong (e.g.  if the aws_volume doesn't exist go merrily along).  I see why it is right, but i think it could be confusing. Could we add some comments here to describe what is going on?